app/src/main/java/com/tictactoe_master/data_to_file/DataToFile.kt

```diff
@@ -0,0 +1,84 @@
+package com.tictactoe_master.data_to_file
+
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import androidx.appcompat.app.AppCompatActivity
+import com.tictactoe_master.R
+
+/*
```

Comment:
This comment is not the best solution to enforce the correct behaviour

You can refactor the write / read functions to check for the context:
```
fun writeInt (context, key, value) {
    if (this.editor == null) {
        // throw Exception("No context given")
        this.setContext(context)
     }
     this.editor!!.putInt(key, value)
}
```

Or you can create a java style singleton with a specified constructor:
```
class FileDataHandler 
    private constructor (private val context: AppCompatActivity) {

    private val sharedPref
    private val editor

    init {
        // set context here
    }

    companion object {
        @Volatile
        private var instance: FileDataHandler? = null

        fun getInstance(context: AppCompatActivity): FileDataHandler = 
            this.instance ?: synchronized(this) {
                instance ?: FileDataHandler(context).also { instance = it }
        }

        fun writeInt (context, key, value) {
            this.getInstance(context).writeInt(key, value)
        }
    }

    fun writeInt (key, value) {
        this.editor.putInt(key, value)
    }
}
```

This way you ensure that the data handler will be used with a not null context